### PR TITLE
[New] `jsx-filename-extension`: add `ignoreFilesWithoutCode` option to allow empty files

### LINKED ISSUE #3644: [jsx-filename-extension]: option to allow empty files
Given the config

```yaml
rules:
  react/jsx-filename-extension: [2, {allow: as-needed]}]
```

a file that is empty or contains only comments triggers the rule, for example a `.jsx` file with:

```js
// Put your components here
```

I think a option to allow empty files when `as-needed` is specified would be ideal for this use case, not sure whether it should be the default, but I think it could be.

### PR DESCRIPTION
**Pull request by burtek**
Fixes #3644

### CODE REVIEW

**Comment by burtek** [comment]
The failed tests are some glitch i runner
image
image

File: lib/rules/jsx-filename-extension.js [hunk 1]
```diff
@@ -97,6 +103,9 @@ module.exports = {
         }
 
         if (isAllowedExtension && allow === 'as-needed') {
+          if (ignoreFilesWithoutCode && node.body.length === 0) {
```
Comment (ljharb) [inline lib/rules/jsx-filename-extension.js:106]: does `node.body` include EmptyStatements, comment nodes, etc?

**Review by ljharb — APPROVED** [review]
LGTM overall

File: lib/rules/jsx-filename-extension.js (hunk 1 above)
Comment (burtek) [reply lib/rules/jsx-filename-extension.js:106]: No, it doesn't

**Pull request merged**